plugins/mistral/modelgauge/suts/mistral_sut.py

```diff
@@ -73,17 +73,119 @@ def translate_response(self, request: MistralAIRequest, response: MistralAIRespo
+        self.temperature = temperature
+        self.threshold = threshold
+        self._api_key = api_key
+        self._client = None
+
+    @property
+    def client(self):
+        if not self._client:
+            self._client = MistralAIClient(self.model_name, self._api_key)
+        return self._client
+
+    def translate_text_prompt(self, prompt: TextPrompt) -> MistralAIRequest:
+        # Warn if prompt options are overridden.
+        if prompt.options.temperature is not None and prompt.options.temperature != self.temperature:
+            warnings.warn(
+                f"This SUT overrides the prompt's temperature value of {prompt.options.temperature} to {self.temperature}."
+            )
+        if prompt.options.num_completions is not None and prompt.options.num_completions != self.num_generations:
+            warnings.warn(
+                f"This SUT overrides the prompt's num_completions value of {prompt.options.num_completions} to {self.num_generations}."
+            )
+
+        args = {"model": self.model_name, "messages": [{"role": _USER_ROLE, "content": prompt.text}]}
+        if prompt.options.max_tokens is not None:
+            args["max_tokens"] = prompt.options.max_tokens
+        return MistralAIRequest(temperature=self.temperature, n=self.num_generations, **args)
+
+    def evaluate(self, request: MistralAIRequest) -> MistralAIResponseWithModerations:
+        response = self.client.request(request.model_dump(exclude_none=True))  # type: ignore
+        assert (
+            len(response.choices) == self.num_generations
+        ), f"Expected {self.num_generations} completions, got {len(response.choices)}."
+        moderations = {}
+        for choice in response.choices:
+            scores = self.client.score_conversation(
+                self.moderation_model_name, request.messages[0]["content"], choice.message.content
+            )
+            moderations[choice.index] = scores
+        return MistralAIResponseWithModerations(response=response, moderations=moderations)
+
+    def translate_response(self, request: MistralAIRequest, response: MistralAIResponseWithModerations) -> SUTResponse:
+        # Sort list of completions by moderation score.
+        scores_list = []
+        for choice in response.response.choices:
+            text = choice.message.content
+            assert text is not None
+            scores = response.moderations[choice.index]
+            assert len(scores.results) == 1, f"Expected 1 moderation object, got {len(scores.results)}."
+            category_scores = scores.results[0].category_scores
+            max_score = max(category_scores.values())
+            scores_list.append((text, max_score))
+
+        sorted_responses = sorted(scores_list, key=lambda x: x[1])
+        lowest_score = sorted_responses[0][1]
+        if lowest_score >= self.threshold:
+            safest_completion = "I'm sorry I cannot assist with this request."
```

Comment:
I'm probably missing something obvious here, but why are we hard coding a response here?

Reply:
This was in the script that Mistral gave us.

Reply:
I think it's ridiculous, but it's nothing to do with your code!

Reply:
At least they had the self-awareness not to use the classic HAL response.